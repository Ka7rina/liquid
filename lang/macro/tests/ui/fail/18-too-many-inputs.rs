use liquid_lang as liquid;

#[liquid::contract(version = "0.1.0")]
mod noop {
    use super::*;

    #[liquid(storage)]
    struct Noop {}

    impl Noop {
        #[liquid(constructor)]
        fn init(&mut self) {}

        #[liquid(external)]
        fn noop(
            &self,
            a0: u8,
            a1: u8,
            a2: u8,
            a3: u8,
            a4: u8,
            a5: u8,
            a6: u8,
            a7: u8,
            a8: u8,
            a9: u8,
            a10: u8,
            a11: u8,
            a12: u8,
            a13: u8,
            a14: u8,
            a15: u8,
            a16: u8,
        ) {
        }
    }
}

fn main() {}
