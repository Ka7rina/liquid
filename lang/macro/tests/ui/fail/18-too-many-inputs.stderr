error: the number of input arguments should not be more than 16
  --> $DIR/18-too-many-inputs.rs:17:13
   |
17 | /             a0: u8,
18 | |             a1: u8,
19 | |             a2: u8,
20 | |             a3: u8,
...  |
32 | |             a15: u8,
33 | |             a16: u8,
   | |___________________^
